lib/keyrod/oidc_client.rb

```diff
@@ -0,0 +1,34 @@
+require 'faraday'
+require 'json'
+
+module Keyrod
+  class OIDCClient
+    def initialize
+      @oidc_endpoint = Keyrod::Settings['oidc-site']
+      @refresh_token = Keyrod::Settings['refresh-token']
+      @client_id = Keyrod::Settings['client-id']
+      @client_secret = Keyrod::Settings['client-secret']
+    end
+
+    def access_token
+      ssl = { verify: Keyrod::Settings['verify-ssl'] }
+      ssl['ca_path'] = Keyrod::Settings['ca-dir'] if Keyrod::Settings['ca-dir']
+      conn = Faraday.new(@oidc_endpoint, ssl: ssl) do |faraday|
+        faraday.request  :url_encoded
+        faraday.response :logger if Keyrod::Settings['debug']
+        faraday.adapter  Faraday.default_adapter
+      end
+
+      response = conn.post do |req|
```

Comment:
Are you relly supposed to use POST and then encode parameters in the URL? It can be done just it seems a bit weird to me.

Reply:
According to http://openid.net/specs/openid-connect-core-1_0.html#RefreshingAccessToken , the request should be POST and should use form serialization

Reply:
:+1: